src/libraries/System.Numerics.Tensors/src/System/Numerics/Tensors/netcore/SpanNDDebugView.cs

```diff
@@ -0,0 +1,30 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace System.Numerics.Tensors
+{
+    internal sealed class SpanNDDebugView<T>
+    {
+        private readonly T[] _array;
+
+        public SpanNDDebugView(SpanND<T> span)
+        {
+            _array = span.ToArray();
+        }
+
+        public SpanNDDebugView(ReadOnlySpanND<T> span)
+        {
+            _array = span.ToArray();
+        }
+
+        [DebuggerBrowsable(DebuggerBrowsableState.RootHidden)]
+        public T[] Items => _array;
```

Comment:
It would be better if we could visualize it as an N dimensional array, but since we don't know the dimensions till runtime I couldn't figure out how to do that so I went this route. I could also have it just display a string output with the right dimensions if that would be better.

Reply:
You could make it a jagged array, e.g. just having an object[] that you then populate with other object[]s which are in turn populated with other object[]s, etc.  Not sure that'd be particularly user friendly.  Or possibly you could special-case known common shapes, e.g. special-case T[,], T[,,], and T[,,,].